core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -287,6 +299,45 @@ oppia.controller('StateInteraction', [
                 stateCustomizationArgsService.displayed = {};
               };
 
+              $scope.isSaveInteractionEnabled = function() {
+                return $scope.hasCustomizationArgs &&
+                  $scope.stateInteractionIdService.displayed &&
+                  $scope.form.schemaForm.$valid &&
+                  $scope.areCustomizationArgsValid();
+              };
+
+              // This function assumes that there are customization args and so
+              // does not do any checks for it.
+              $scope.areCustomizationArgsValid = function() {
```

Comment:
Why does this need to be a separate function? Could it be combined with the previous one?